graph/appender/sidecars_check.go

```diff
@@ -34,7 +34,7 @@ func (a *SidecarsCheckAppender) applySidecarsChecks(trafficMap graph.TrafficMap,
 	for _, n := range trafficMap {
 
 		// Skip the check on the sidecars if we don't have access to the namespace
-		if _, found := a.AccessibleNamespaces[n.Namespace]; !found {
+		if _, found := a.Namespaces[n.Namespace]; !found {
```

Comment:
I think there is a mistake here.  Maybe my original comment was confusing.  When I suggested not adding duration information to AccessibleNamespaces I did not mean remove AccessibleNamespaces.  I think you need to keep the use of AccessibleNamespaces as it was before.  There are two things going on.  First there are the namespaces requested for the graph, say A and B.   Then there are the AccessibleNamespaces for the user, say A, B, C.

First, if a requested namespace is not in the set of accessible namespaces the request should fail fast in the Options processing.  But remember now that when we generate a graph for A and B it can have leaf nodes outside of A and B, those nodes either request services in A or B, or are requested by services in A or B.  For those outsider nodes we need to know, in various places, if the user has full access.  This is one of those places.

So, basically, a,Namespaces deals with the *requested* namespaces (all of which must be accessible), but is only a subset of AccessibleNamespaces.

Reply:
No, I misunderstood the usage of `AccessibleNamespaces`. I thought that both will have the same items, but `AccessibleNamespaces` can have more items. I'll fix this.